### ISSUE TITLE
Got inf loss when training Chinese CRNN with fp32

### ISSUE THREAD

**Issue by hqkate** [mindspore-lab/mindocr#283]
With amp_level=O3, as in [tools/train.py](https://github.com/mindspore-lab/mindocr/blob/main/tools/train.py?plain=1#L102) , `pred_cast_fp32` equals to True, which means the input logit of CTCLoss will be casted into fp32. This will cause the inf problem as shown in [debug_loss_inf_castfp32.log](https://github.com/mindspore-lab/mindocr/files/11460094/debug_loss_inf_castfp32.log)

If set `pred_cast_fp32` equals to False, the loss won't be inf. Please see [debug_loss_inf_nocast.log](https://github.com/mindspore-lab/mindocr/files/11460097/debug_loss_inf_nocast.log)

**Comment by SamitHuang** [comment]
amp为O2, O3时，网络prediction输出为fp16, 不转的话det loss计算会出错

**Comment by zhtmike** [comment]
那是否预设pred_cast_fp32 = False 更好? 当loss需要fp32 input再改为True

**Comment by SamitHuang** [comment]
https://github.com/mindspore-lab/mindocr/pull/284
